### DISCUSSION TITLE
Looking to fetch dns and udp connections

### DISCUSSION THREAD

**Discussion by Engg-blogger** [bpftrace/bpftrace#2668] [General]
I am new to this tool and had a question about how to only fetch dns connections. When I looked at the bcc tools, it has the support to show dns connection under tcpconnect tool by passing -d as parameter. However, tcpconnect tool in bpf tracing doesnt have any mention of dns. Also is there any tool that can provide udp connections?

Here is the link to the bcc tool : https://github.com/iovisor/bcc/blob/master/tools/tcpconnect.py

**Comment by ajor** [comment]
The bpftrace tcpconnect doesn't have support for tracing DNS or UDP. If you want to look at them you'll have to write your own script.

Take a look at how the BCC tool traces DNS/UDP and it shouldn't be too hard to replicate that in bpftrace: https://github.com/iovisor/bcc/blob/8422cd449ad2e60414a4508aa4a150a9db39c4a3/tools/tcpconnect.py#L535

**Reply by Engg-blogger** [reply]
Thanks. Are there any guidelines on how to write a new tool. Example tool creation from bcc tool will be helpful.

**Reply by ajor** [reply]
There's nothing special about bpftrace tools - they're just a selection of useful pre-written bpftrace scripts. There's some documentation on writing bpftrace scripts [here](https://github.com/iovisor/bpftrace/blob/master/docs/tutorial_one_liners.md) and [here](https://github.com/iovisor/bpftrace/blob/master/docs/reference_guide.md).

**Comment by Engg-blogger** [comment]
I have one more question which is related to tcp_connect.bt. It says it is only limited to ipv4 at [https://github.com/iovisor/bpftrace/blob/master/tools/tcpconnect.bt#L9]
But here [https://github.com/iovisor/bpftrace/blob/master/tools/tcpconnect.bt#L9](url) it is checking for both ipv4 and v6. Could you elaborate whether this tool supports both ipv4 and v6 or just ipv4?

**Reply by fbs** [reply]
The comment was never updated, ipv6 should work :)

**Comment by Engg-blogger** [comment]
Thanks... I have one more question. So I might be using this tool on various distros. Is there a script that can run on any distro to download the corresponding binary for that distro?

**Reply by viktormalik** [reply]
You can download the pre-built binary that we provide, e.g. for the latest 0.18 version from [here](https://github.com/iovisor/bpftrace/releases/download/v0.18.0/bpftrace). Note that it's for x86 only, though.
